crates/noirc_evaluator/src/ssa_refactor/abi_gen/mod.rs

```diff
@@ -1,29 +1,36 @@
 use std::collections::BTreeMap;
 
 use acvm::acir::native_types::Witness;
-use iter_extended::{btree_map, vecmap};
+use iter_extended::btree_map;
 use noirc_abi::{Abi, AbiParameter, FunctionSignature};
 
 /// Arranges a function signature and a generated circuit's return witnesses into a
 /// `noirc_abi::Abi`.
-pub(crate) fn gen_abi(func_sig: FunctionSignature, return_witnesses: Vec<Witness>) -> Abi {
+pub(crate) fn gen_abi(
+    func_sig: FunctionSignature,
+    return_witnesses: Vec<Witness>,
+    input_witnesses: &[Witness],
```

Comment:
Can you reorder the parameters here so that the input witnesses are before the return witnesses?

Reply:
Done